### PR TITLE
Natalie branch

### PR DESCRIPTION
**Pull request by natalie-y-kim**
adding trained ResNet model weight

### CODE REVIEW

**Review by akshayratnawat — APPROVED** [review]

File: resnet_train.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+# Unfreeze last two layers: layer4 and fc
+for param in model.layer4.parameters():
+    param.requires_grad = True
+
+for param in model.fc.parameters():
+    param.requires_grad = True
+
+model.fc = nn.Linear(model.fc.in_features, 2)  # 2 classes (real, fake)
+
```
Comment (akshayratnawat) [inline resnet_train.py:50]: You unfreeze the old fc layer, then immediately overwrite it with a new nn.Linear layer.
That means the requires_grad settings you applied are lost.

File: resnet_train.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        images, labels = images.to(device), labels.to(device)
+        optimizer.zero_grad()
+
+        outputs = model(images)
+        loss = criterion(outputs, labels)
+        loss.backward()
+        optimizer.step()
+
+        train_loss += loss.item()
+        correct += (outputs.argmax(1) == labels).sum().item()
```
Comment (akshayratnawat) [inline resnet_train.py:75]: train_loss += loss.item()
This sums the loss per batch. When you print it, the scale depends on the number of batches.

Usually, you report average loss per batch or per sample.

File: resnet_train.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+    acc = correct / len(train_loader.dataset) * 100
+    print(f"Epoch {epoch+1}, Loss: {train_loss:.4f}, Accuracy: {acc:.2f}%")
+
+    train_losses.append(train_loss)
+    train_accuracies.append(acc)
+
+
+# save model
+torch.save(model, 'resnet18_full_model.pth')
```
Comment (akshayratnawat) [inline resnet_train.py:85]: This saves the entire model object, which can break if you load it in a different PyTorch version.
Safer approach is to save only state_dict().

File: resnet_train.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], 
+                         std=[0.229, 0.224, 0.225])
+])
+
+def rgba_loader(path):
+    img = default_loader(path)
+    if img.mode == 'P' or img.mode == 'LA':
+        img = img.convert('RGBA')
+    return img
+
```
Comment (akshayratnawat) [inline resnet_train.py:27]: You convert some images to RGBA, but ResNet18 expects 3 channels (RGB).
After .convert("RGBA"), your ToTensor() will return 4 channels → mismatch with ResNet.

File: resnet_train.py (hunk 2 above)
Comment (akshayratnawat) [inline resnet_train.py:75]: You are only tracking train accuracy. Add a test loop accuracy to monitor overfitting.
